File: Practice(Beginner)/C++/The Smallest Pair.cpp
// Source: https://www.codechef.com/viewsolution/29120960

#include <bits/stdc++.h>
using namespace std;
int main() {
    long long int t, n, temp;
    cin >> t;
    while (t--){
        cin >> n;
        vector<long long int> arr;
        for (int i = 0; i < n; ++i) {
            cin >> temp;
            arr.push_back(temp);
        }
        sort(arr.begin(),arr.end());
        cout << arr[0] + arr [1] << endl;
    }
    return 0;
}

// Alternate way

/*

#include <bits/stdc++.h>
using namespace std;
int main() {
    int t, n;
    cin >> t;
    while (t--){
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; ++i) {
            cin >> arr[i];
        }
        int minVal = INT_MAX;
        for (int i = 0; i < n-1; ++i) {
            minVal = min(minVal,(arr[i] + arr[i+1]));
        }
        cout << minVal << endl;
    }
    return 0;
}

*/

File: Practice(Beginner)/Python3/The_Smallest_Pair.py
# Source: https://www.codechef.com/viewsolution/29120960

n = int(input())
for j in range(n):
    x = int(input())
    l = list(map(int, input().split()))
    s = l[0]+l[1]
    for i in range(len(l)-1):
        if(s > l[i]+l[i+1]):
            s = l[i]+l[i+1]
    print(s)


# Alternate way

"""
n = int(input())
for j in range(n):
    x = int(input())
    l = list(map(int, input().split()))
    l.sort()
    print(l[0]+l[1])
"""